db/db_iter.cc

```diff
@@ -306,7 +307,7 @@ class DBIter final: public Iterator {
   const SliceTransform* prefix_extractor_;
   Env* const env_;
   Logger* logger_;
-  const Comparator* const user_comparator_;
```

Comment:
Here "const Comparator* const" is reduced to "const Comparator*". Could it disable the compiler form doing some optimizations?

Reply:
I'm not sure about compiler optimization. The change is because `InternalComparator` needs to be copyable, and so any of its member cannot have constant member.

Reply:
Are you sure? I applied this patch and it compiled fine!
```
diff --git a/util/user_comparator_wrapper.h b/util/user_comparator_wrapper.h
index 43797709c..67fb94c34 100644
--- a/util/user_comparator_wrapper.h
+++ b/util/user_comparator_wrapper.h
@@ -59,7 +59,7 @@ class UserComparatorWrapper final : public Comparator {
   }

  private:
-  const Comparator* user_comparator_;
+  const Comparator* const user_comparator_;
 };

 }  // namespace rocksdb

Reply:
have you try compiling db_memtable_test?

Reply:
Oh you are right.